src/trace-writer.js

```diff
@@ -256,7 +256,7 @@ TraceWriter.prototype.publish_ = function(projectId, json) {
   }, function(err, response, body) {
     if (err) {
       that.logger_.error('TraceWriter: error: ',
-        (response && response.statusCode) || '', err);
+        (response && response.statusCode) || '' + '\n' +  err.message + '\n' + err.stack);
```

Comment:
I think you need parentheses around `(response && response.statusCode) || ''` if a stack trace should be reported regardless of whether there was a response or not.